ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -302,7 +302,7 @@ - (instancetype)initWithTask:(id<ORKTask>)task restorationData:(NSData *)data de
     
     self = [self initWithTask:task taskRunUUID:nil];
     
-    if (self) {
+    if (self && (data != nil)) {
         self.delegate = delegate;
```

Comment:
Should accept `delegate` regardless of restoration. 
Shall we move `self.delegate = delegate;` out of `if block`?